src/include/mpir_hw_topo.h

```diff
@@ -0,0 +1,189 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef MPIR_HW_TOPO_H_INCLUDED
+#define MPIR_HW_TOPO_H_INCLUDED
+
+/*
+ * Tracks hwloc_obj_type_t do not change ordering
```

Comment:
Why is the ordering important?

Reply:
I just wanted to have the possibility to pass these directly to `hwloc` without checking if `MPIR_HWLOC_OBJ_MACHINE` is actually equal to `HWLOC_OBJ_MACHINE` and doing the conversion if not.